os/board/rtl8730e/src/component/soc/amebad2/fwlib/ram_common/ameba_i2c.c

```diff
@@ -717,6 +718,14 @@ u32 I2C_MasterRead(I2C_TypeDef *I2Cx, u8 *pBuf, u32 len)
 				I2C_ClearAllINT(I2Cx);
 				return cnt;
 			}
+			DelayUs(1);
+			if (timeout == 0) {
+				RTK_LOGW(TAG, "I2C_MasterRead timeout - resetting I2C\n");
+				I2C_Cmd(I2Cx, DISABLE);
+				I2C_Cmd(I2Cx, ENABLE);
```

Comment:
When all datas are read, you don't call these disable and enable. Is this necessary when we don't get full data only? If we call this, what happen?

Reply:
We don't have to call it when all data are read, this is normally called when we want to change slave addr, or when it enter a trap state. When we are stuck in waiting fifo flag, it is more likely that it enter a trap state so required a I2C reset, so to prepare for next read. 
But even if we call it when getting all data, it should not create any issue.